programs/utilities/lockRotation.sc

```diff
@@ -0,0 +1,347 @@
+	rayhit  = query(p, 'trace', 5, 'blocks');
+    if (raycast != null,
+        hit_dir = _castSide(rayhit, raycast);
+		
+        half = abs(floor((raycast:1 - floor(raycast:1)) * 2));
+        if (_dirAxis(direction) == 'y',
+            half = -1;
+        , raycast:1 < 0,
+            half = 1 - half;
+        );
+        
+		flat = _flipDir(hit_dir);
+		if (_contains(['up', 'down'], flat), flat = flat_dir);
+		
+        newProfile = [hit_dir, half, _flipDir(flat)];
+    ,
+        newProfile = [direction, -1, flat_dir];
+    );
+    
+    if (global_lockProfile != null && newProfile == global_lockProfile,
+        global_lockProfile = null;
+    ,
+		_displayProfile(raycast, rayhit, newProfile);
+        global_lockProfile = newProfile;
+    );
+);
+_getIdealState(block) ->
+(
+    state = block_state(block);
+    if(_isBustable(block), return(state));
+    
+    direction = global_lockProfile:0;
+    half      = global_lockProfile:1;
+    flat_dir  = global_lockProfile:2;
+    if (global_lockProfile == null, return(state));
+    
+    if (_contains(state, 'axis'), state:'axis' = _dirAxis(direction));
+    if (_contains(state, 'type'),
+        //Slab Type
+        if (_contains(['top', 'bottom'], state:'type') && half != -1,
+            state:'type' = _halfSide(half);
+        );
+    );
+    if (_contains(state, 'facing'),
+        state:'facing' = flat_dir; //Generic
+        //Hopper exception
+        if (block == 'hopper',
+            state:'facing' = _flipDir(flat_dir);
+            if (direction == 'up', state:'facing' = 'down');
+        );
+        //Vertibles exception
+        if (_isVertible(block),
+            state:'facing' = direction;
+            //Grindstone handling
+            if (block == 'grindstone',
+                state:'face' = 'wall';
+                if (direction == 'up', state:'face' = 'floor');
+                if (direction == 'down', state:'face' = 'ceiling');
+            );
+        );
+        //Stairs addendum
+        if (_contains(state, 'half'),
+            if (_contains(['top', 'bottom'], state:'half'),
+				if (state:'open' == null, state:'facing' = _flipDir(flat_dir));
+                if (half != -1, state:'half' = _halfSide(half));
+            );
+        );
+    );
+    
+    return(state);
+);
+                                                                               
+__on_player_places_block(player, item_tuple, hand, block) ->
+(
+    properties = _getIdealState(block);
+    set(pos(block), block, properties);
+	if (_contains(block, 'stairs'), global_rotated = block);
+);
+__on_tick() ->
+(
+	if(global_rotated != null,
+		//sleep(1000);
+		//for(neighbours(global_rotated),
+		//	print(pos(_));
+		//	update(pos(_));
+		//);
+		//update(pos(global_rotated));
+		upos = [pos(global_rotated):0 + 1, pos(global_rotated):1, pos(global_rotated):2];
+		ublock = block(upos);
+		set(upos, 'air');
+		place_item('stone', upos);
+		set(upos, ublock);
+		global_rotated = null;
```

Comment:
Seen as though line 329 is the only place where the `global_rotated` variable becomes not null, can't you just move the code from `on_tick` into the if statement?
Or if you need a delay, use `schedule()`?

Reply:
That snippet of code went through a lot of trial and error, and I basically stopped touching it once it worked. You're correct though, after testing it appears that code can, in fact, be moved to the if statement. I've made that change now.